e2etest/bgp_container.go

```diff
@@ -0,0 +1,189 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+Copyright 2021 The MetalLB Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+// https://github.com/kubernetes/kubernetes/blob/92aff21558831b829fbc8cbca3d52edc80c01aa3/test/e2e/network/loadbalancer.go#L878
+
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"html/template"
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/pkg/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientset "k8s.io/client-go/kubernetes"
+)
+
+const (
+	// BGPTestFRRConfigDir contains the name of the BGP configuration directory.
+	BGPTestFRRConfigDir = "config/frr"
+	// BGPDConfigFileName contains the BGP router configuration file.
+	BGPDConfigFileName = "bgpd.conf"
+	// FRRRoutingImage contains the name and tag of the FRR routing image.
+	FRRRoutingImage = "frrouting/frr:v7.5.1"
+	// FRRContainerMountDestination contains the mount destination path.
+	FRRContainerMountDestination = "/etc/frr"
+)
+
+// Run a BGP router in a container.
+// Returns the IPv4 and IPv6 addresses of the container.
+func startFRRContainer(cs clientset.Interface, containerName string, testDirName string, bgpConfig string) (string, string, error) {
```

Comment:
I would let this just start the container with an empty config file, and implement another method `reloadFRRConfig` that basically leverages what @oribon used in https://github.com/metallb/metallb/pull/935 .
This will make handling external containers more flexible, and will also make it easier to use the same set of tests when we are running frr on localhost.